client/src/partials/exchange/modal.js

```diff
@@ -27,10 +39,24 @@ function ExchangeModalController(Rates, Currencies, Session, $uibModalInstance,
 
   function submit(invalid) {
     if (invalid) { return; }
+    vm.data.enterprise_id = vm.enterpriseCurrencyId;
```

Comment:
That is a mistake.  `enteprise_id` should be the enterprise id.  Why would you name a column one name and store completely different data in it?